[Doc] fix format configuration

DanRoscigno opened the pull request:
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

DanRoscigno committed:
init

Signed-off-by: DanRoscigno <dan@roscigno.com>

DanRoscigno committed:
replace tables with H5 and list

Signed-off-by: DanRoscigno <dan@roscigno.com>

DanRoscigno committed:
fix linting

Signed-off-by: DanRoscigno <dan@roscigno.com>

DanRoscigno committed:
fix indentation

Signed-off-by: DanRoscigno <dan@roscigno.com>

DanRoscigno committed:
fix indentation

Signed-off-by: DanRoscigno <dan@roscigno.com>

wanpengfei-git commented:
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: s
+- **Default**: 300
+- **Description**: The maximum duration by which the metadata on the follower and observer FEs can lag behind that on the leader FE. Unit: seconds. If this duration is exceeded, the non-leader FE stops providing services.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/37abe88c-a38b-45f8-ae2b-13751843ab27)
-> the non-leader FEs stop
because we have follower and observer FEs in addition to one Leader FE in a cluster, and that solicits a plural form for "non-leader FEs".

amber-create replied:
![image](https://github.com/StarRocks/starrocks/assets/57167462/23c39226-9c0f-49bd-97b9-52826ddd9e30)
"Unit: seconds." can be deleted, because the unit has been declared above in Line 85 "**Unit**: s".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: -
+- **Default**: FALSE
+- **Description**: Whether to view the profile of a query. If this parameter is set to `TRUE`, the system collects the profile of the query. If this parameter is set to `FALSE`, the system does not collect the profile of the query.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/05fdeda3-8a0d-4d4e-a04d-74318776662f)
I think it's better to use "collect" compared to "view", 'cause "collect" is an action done by the system while "view" is an action done by the user and clearly here it refers an action done by the system.

DanRoscigno replied:
```suggestion
- **Description**: Whether to collect the profile of a query. If this parameter is set to `TRUE`, the system collects the profile of the query. If this parameter is set to `FALSE`, the system does not collect the profile of the query.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: s
+- Default: 600
+- Description: The maximum timeout duration for creating a table, in seconds.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/0118afb4-2152-412d-b42b-60f2fec24ab0)
", in seconds" can be deleted.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: LONG
+- Default: 107374182400
+- Description: The size of the largest partition for automatic collection to collect data. Unit: Byte. If a partition exceeds this value, full collection is discarded and sampled collection is performed instead.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/908943d2-2ea5-4bcc-974c-eafa83318a9b)
1. Which unit is correct?
2. Remove the unit in "Description".

DanRoscigno replied:
yes, this is confusing.  The unit is a Long, but it represents the number of Bytes.  I will reword the sentence.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: INT
+- Default: 5000000000
+- Description: The maximum number of rows to query for a single analyze task. An analyze task will be split into multiple queries if this value is exceeded.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/163a625c-87f7-43f3-ad89-caa794e21a0f)
"INT" should be a data type I think.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: s
+- Default: 300
+- Description: The interval for checking data updates during automatic collection. Unit: seconds.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/98814274-b025-41ca-8368-927eb4598b53)
Delete "Unit: seconds."

DanRoscigno replied:
```suggestion
- Description: The interval for checking data updates during automatic collection.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: STRING
+- Default: 23:59:59
+- Description: The end time of automatic collection. Value range: `00:00:00` - `23:59:59`.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/1a346103-4bfb-4abc-90cf-2b267bbdbce4)
1. "Unit: STRING" in Line 231 and Line 237 should be "Type: STRING" I think.
2. I suggest moving "Value range: xxx" up preceding "Default", so the flow becomes "Type" -> "Value range" -> "Default" and then the "Description".

DanRoscigno replied:
we need to defer this and have RD comment.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: s
+- Default: 60
+- Description: The interval at which metadata is scheduled. Unit: seconds. The system performs the following operations based on this interval:
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/a60d0afe-d19d-4680-883f-4b25699b8785)
Remove "Unit: seconds."

DanRoscigno replied:
```suggestion
- Description: The interval at which metadata is scheduled. The system performs the following operations based on this interval:
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+- Default: 60
+- Description: The interval at which metadata is scheduled. Unit: seconds. The system performs the following operations based on this interval:
+  - Create tables for storing statistics.
+  - Delete statistics that have been deleted.
```
I think the two "delete" refer to different operations in different contexts, but can't see what the differences are. Is there any more details that we can give?

DanRoscigno replied:
Defer

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: s
+- Default: `24 * 60 * 60`
+- Description: The interval at which the cache of statistical information is updated. Unit: seconds.
```
Delete "Unit: seconds."

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: s
+- Default: 259200
+- Description: The duration to retain the history of collection tasks. The default value is 3 days. Unit: seconds.
```
Delete "Unit: seconds."

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: -
+- Default: 3
+- Description: The maximum number of manual collection tasks that can run in parallel. The value defaults to 3, which means you can run a maximum of three manual collections tasks in parallel. If the value is exceeded, incoming tasks will be in the PENDING state, waiting to be scheduled.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/350d7a1d-dad7-428d-98a6-c57acbebf982)
collections -> collection

amber-create replied:
![image](https://github.com/StarRocks/starrocks/assets/57167462/d0cd373a-e21d-4d3a-8491-854959762f4c)
Wrap the parameter value in code format would be best.

DanRoscigno replied:
```suggestion
- Description: The maximum number of manual collection tasks that can run in parallel. The value defaults to 3, which means you can run a maximum of three manual collection tasks in parallel. If the value is exceeded, incoming tasks will be in the PENDING state, waiting to be scheduled.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: -
+- Default: FALSE
+- Description: Whether to select local replicas for queries. Local replicas reduce the network transmission cost. If this parameter is set to TRUE, the CBO preferentially selects tablet replicas on BEs that have the same IP address as the current FE. If this parameter is set to FALSE, both local replicas and non-local replicas can be selected. The default value is FALSE.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/1d75d09d-5bbb-4aad-8f52-1ee2e6335b01)
Wrap the parameter values in code format.

amber-create replied:
![image](https://github.com/StarRocks/starrocks/assets/57167462/7a33c1ce-ab6a-404b-ac68-d65c505b15ad)
Delete "The default value is FALSE.", because the "Default" is already declared in Line 301 "Default: FALSE".

DanRoscigno replied:
```suggestion
- Description: Whether to select local replicas for queries. Local replicas reduce the network transmission cost. If this parameter is set to TRUE, the CBO preferentially selects tablet replicas on BEs that have the same IP address as the current FE. If this parameter is set to `FALSE`, both local replicas and non-local replicas can be selected. The default value is FALSE.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: s
+- **Default**: 300
+- **Description**: The maximum loading lag that can be tolerated by a BE replica. If this value is exceeded, cloning is performed to clone data from other replicas. Unit: seconds.
```
Delete "Unit: seconds."

DanRoscigno replied:
```suggestion
- **Description**: The maximum loading lag that can be tolerated by a BE replica. If this value is exceeded, cloning is performed to clone data from other replicas.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: s
+- **Default**: 259200
+- **Description**: The maximum timeout duration allowed for a load job. The load job fails if this limit is exceeded. This limit applies to all types of load jobs. Unit: seconds.
```
Delete "Unit: seconds."

DanRoscigno replied:
```suggestion
- **Description**: The maximum timeout duration allowed for a load job. The load job fails if this limit is exceeded. This limit applies to all types of load jobs.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: s
+- **Default**: 259200
+- **Description**: The maximum duration the labels of load jobs that have been completed and are in the FINISHED or CANCELLED state can be retained in the StarRocks system. The default value is 3 days. After this duration expires, the labels will be deleted. This parameter applies to all types of load jobs. Unit: seconds. A value too large consumes a lot of memory.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/f2cabfb6-91a8-433e-8b76-5ca1517f2826)
I think something is missing between "The maximum duration" and "the labels". Probably we can add "for which" between them so the sentence reads "The maximum duration for which the labels ... can be retained".

amber-create replied:
![image](https://github.com/StarRocks/starrocks/assets/57167462/4224bb67-c616-41df-974a-a39d0378b7e8)
I'm not sure if "The default value is 3 days." is clear and won't solicit ambiguity, because the unit is "s" and "the value" can't be "3 days" but a much longer integer measured in seconds. So, I think it's better to say, for example, "The default maximum duration is 3 days."

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+##### max_routine_load_batch_size
+
+- **Unit**: Byte
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/da2a8252-3aba-45d6-ac57-474e6a816bdc)
I think "Unit: Byte" should be "Unit: bytes". Also, please remove ", in bytes".

DanRoscigno replied:
defer

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: Min
+- **Default**: 5
+- **Description**: The interval at which Routine Load jobs are automatically recovered, in minutes.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/745d84af-311c-48c8-b038-8b5de6428b3a)
See comments made earlier

DanRoscigno replied:
```suggestion
- **Description**: The interval at which Routine Load jobs are automatically recovered.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: Byte
+- **Default**: 268435456
+- **Description**: The maximum amount of data that can be exported from a single BE by a single data unload task, in bytes.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/47ed262f-1fcc-4362-9acb-7d72924857a1)
Please search for all "in bytes" starting from here till the end of this doc and remove them if the Unit is already given preceding the Description?

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: -
+- **Default**: TRUE
+- **Description**: Whether to return an error message "all partitions have no load data" if no data is loaded. Values: TRUE: If no data is loaded, the system displays a failure message and returns an error "all partitions have no load data". FALSE: If no data is loaded, the system displays a success message and returns OK, instead of an error.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/620da93d-393f-413a-be02-04e0dfd5b87b)
I think the part starting from "Values" is supposed to be an unordered list, like below:
Values:
- TRUE: ...
- FALSE: ...

DanRoscigno replied:
```suggestion
- **Description**: Whether to return an error message "all partitions have no load data" if no data is loaded. Values:
  - TRUE: If no data is loaded, the system displays a failure message and returns an error "all partitions have no load data".
  - FALSE: If no data is loaded, the system displays a success message and returns OK, instead of an error.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: ms
+- **Default**: 86400*1000
+- **Description**: The timeout duration of a backup job, in ms. If this value is exceeded, the backup job fails.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/875dd46f-fe06-4c82-bcb5-bb18770f6c15)
Delete ", in ms".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
 
-#### Server
+- **Default:** 1024 MB
+- **Description:** The size per log file. Unit: MB. The default value `1024` specifies the size per log file as 1 GB.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/800e7415-3f30-483b-85a9-544ac20b0e74)
Move "Unit: MB." up and make it a new line preceding "Default".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### log_roll_size_mb
 
-#### Server
+- **Default:** 1024 MB
```
Remove "MB" from "Default:** 1024 MB", as "Default" only declares the default value and we will have a separate line "Unit: MB" to declare the unit is MB.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
 
-#### Query engine
+- **Default:** INFO
+- **Description:** The severity levels into which system log entries are classified. Valid values: `INFO`, `WARN`, `ERROR`, and `FATAL`.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/fdd8a142-f39b-44df-a592-65c71e12f63a)
I think it's better to give the "Valid values" preceding the "Default".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### thrift_client_timeout_ms
+
+- **Default:** 5000
+- **Description:** The length of time after which idle client connections time out. Unit: ms.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/6d014f84-e39c-4ef1-8998-7473e5d38a06)
Move "Unit: ms." up preceding the "Default".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### bdbje_heartbeat_timeout_second
+
+- **Default:** 30
+- **Description:** The amount of time after which the heartbeats among the leader, follower, and observer FEs in the StarRocks cluster time out. Unit: second.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/9edd999b-d5de-44cd-b1b9-5a2e1fee522b)
Remove "Unit: second." in the "Description", and add "Unit: seconds." preceding Line 968 "**Default:** 30".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### max_bdbje_clock_delta_ms
+
+- **Default:** 5000
+- **Description:** The maximum clock offset that is allowed between the leader FE and the follower or observer FEs in the StarRocks cluster. Unit: ms.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/877f1820-ef7b-4acd-874f-d8a611830684)
See the comment at Line 969

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### bdbje_replica_ack_timeout_second
+
+- **Default:** 10
+- **Description:** The maximum amount of time for which the leader FE can wait for ACK messages from a specified number of follower FEs when metadata is written from the leader FE to the follower FEs. Unit: second. If a large amount of metadata is being written, the follower FEs require a long time before they can return ACK messages to the leader FE, causing ACK timeout. In this situation, metad ...
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/00955844-ce7f-4b92-8cb0-20a4a62b0a81)
Can you check for all "Unit:" in "Description" throughout the doc and then remove them from "Description" and add them preceding "Default"?

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+- **Description**: The running mode of the StarRocks cluster. Valid values: shared_data and shared_nothing (Default).
+
+shared_data indicates running StarRocks in shared-data mode.
+shared_nothing indicates running StarRocks in shared-nothing mode.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/344dae45-4a5d-4ec0-bb3d-f8d837a7da9d)
I think it's better to organize this part as follows:
Valid values:
- `shared_data` indicates running StarRocks in shared-data mode.
- `shared_nothing` indicates running StarRocks in shared-nothing mode.

DanRoscigno replied:
```suggestion
  - shared_nothing indicates running StarRocks in shared-nothing mode.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### cloud_native_meta_port
+
+- **Default**: 6090
+- **Description**: The cloud-native meta service RPC port. Default: 6090.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/f19a02fb-6bef-46cf-b8b5-3327643b7b42)
Delete "Default: 6090."

DanRoscigno replied:
```suggestion
- **Description**: The cloud-native meta service RPC port.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### cloud_native_storage_type
+
+- **Default**: S3
+- **Description**: The type of object storage you use. In shared-data mode, StarRocks supports storing data in Azure Blob (supported from v3.1.1 onwards), and object storages that are compatible with the S3 protocol (such as AWS S3, Google GCP, and MinIO). Valid value: S3 (Default) and AZBLOB. If you specify this parameter as S3, you must add the parameters prefixed by aws_s3. If you specify this ...
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/7063cb91-cb88-492b-b29c-6741e945cb9b)
Change "Valid value: S3 (Default) and AZBLOB." to "Valid values: S3, AZBLOB." and move it up preceding "Default".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### aws_s3_path
+
+- **Default**: N/A
+- **Description**: The S3 path used to store data. It consists of the name of your S3 bucket and the sub-path (if any) under it, for example, testbucket/subpath.
```
Wrap "testbucket/subpath" in code format

DanRoscigno replied:
```suggestion
- **Description**: The S3 path used to store data. It consists of the name of your S3 bucket and the sub-path (if any) under it, for example, `testbucket/subpath`.
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### aws_s3_region
+
+- **Default**: N/A
+- **Description**: The region in which your S3 bucket resides, for example, us-west-2.
```
Wrap "us-west-2" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### aws_s3_use_aws_sdk_default_behavior
+
+- **Default**: false
+- **Description**: Whether to use the default authentication credential of AWS SDK. Valid values: true and false (Default).
```
Remove "Valid values: true and false (Default)."

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### aws_s3_use_instance_profile
+
+- **Default**: false
+- **Description**: Whether to use Instance Profile and Assumed Role as credential methods for accessing S3. Valid values: true and false (Default).
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/00f95a24-c24a-43c2-a000-0d8137462b0b)
Remove "Valid values: true and false (Default).".
I'm thinking about whether we should add "Valid values: true, false" or "Type: BOOLEAN" for such types of parameters. What's your opinion?

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+  - If you use IAM user-based credential (Access Key and Secret Key) to access S3, you must specify this item as false, and specify aws_s3_access_key and aws_s3_secret_key.
+  - If you use Instance Profile to access S3, you must specify this item as true.
+  - If you use Assumed Role to access S3, you must specify this item as true, and specify aws_s3_iam_role_arn.
+  - And if you use an external AWS account, you must also specify aws_s3_external_id.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/846be8fb-7c56-4d78-a9e6-adb7ef57cab8)
I suggest wrapping all these in code format

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+##### azure_blob_path
+
+- **Default**: N/A
+- **Description**: The Azure Blob Storage path used to store data. It consists of the name of the container within your storage account and the sub-path (if any) under the container, for example, testcontainer/subpath.
```
Wrap "testcontainer/subpath" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+#### report_task_interval_seconds
+
+- **Default:** 10 seconds
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/fb678dd5-808b-4402-8b7e-82cd74acf19d)
Change "**Default:** 10 seconds" to "**Default:** 10", and add "Unit: seconds" preceding "Default", I think.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+#### max_pushdown_conditions_per_column
+
+- **Default:** 1,024 (Maximum Number of Pushdown Conditions per Column)
+- **Description:** The maximum number of conditions that allow pushdown in each column. If the number of conditions exceeds this limit, the predicates are not pushed down to the storage layer.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/5204a217-8aa7-4d73-9d8b-60d6860165ad)
All these should be deleted, I think.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+#### storage_flood_stage_usage_percent
+
+- **Default:** 95% (Storage Usage Threshold for Load and Restore Job Rejection)
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/8d653a15-805b-46db-86fe-ddf882885606)
All these should be deleted.

DanRoscigno replied:
```suggestion
- **Default:** 95
```

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+#### be_exit_after_disk_write_hang_second
+
+- **Default:** 60 seconds
+- **Description:** The length of time that the BE waits to exit after the disk hangs.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/a1672d21-e020-498a-b151-00f1ee9066ed)
All these should be deleted.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+#### size_tiered_level_num
+
+- **Default:** 7 (Number of Levels for Size-tiered Compaction)
```
Can you please check all "Default" starting from here till the end of this doc to see if they contain a parameter description wrapped in parentheses, and remove such descriptions as they are unnecessary?

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: false
+- **Unit**: N/A
+- **Description**: Specifies whether to enable the hedged read feature. This parameter is supported from v3.0 onwards.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/b0f93985-04ee-4c1c-97e1-32cdfbbcc416)
In previous sections, "Unit" precedes "Default". I think it's better to adhere to one sequence "Unit -> Default" throughout this doc. Please see to all the parameters starting from here till the end of this doc.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+#### hdfs_client_hedged_read_threshold_millis
+
+- **Default**: 2500
+- **Unit**: Millisecond
```
"Millisecond" VS "ms", I think it's better to adhere to one style throughout this doc.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: 2500
+- **Unit**: Millisecond
+- **Description**: Specifies the number of milliseconds to wait before starting up a hedged read. For example, you have set this parameter to 30. In this situation, if a read from a block has not returned within 30 milliseconds, your HDFS client immediately starts up a new read against a different block replica. This parameter is supported from v3.0 onwards. It is equivalent to the dfs.client.hed ...
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/e4002b28-1c9e-4fd8-9c57-8333f9a93b1f)
Wrap "30" in code format

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: -1
+- **Unit**: N/A
+- **Description**: The number of bthreads of a BRPC. The value -1 indicates the same number with the CPU threads.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/67ec571f-563f-4a4b-8832-cf166d7dc0d3)
Warp "-1" in code format

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: Empty string
+- **Unit**: N/A
+- **Description**: The module of the logs to be printed. For example, if you set this configuration item to OLAP, StarRocks only prints the logs of the OLAP module. Valid values are namespaces in BE, including starrocks, starrocks::debug, starrocks::fs, starrocks::io, starrocks::lake, starrocks::pipeline, starrocks::query_cache, starrocks::stream, and starrocks::workgroup.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/63614deb-7f22-493a-972b-01c7e338a6b2)
It would be better to wrap all these in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: Empty string
+- **Unit**: N/A
+- **Description**: The strategy for flushing logs. The default value indicates that logs are buffered in memory. Valid values are -1 and 0. -1 indicates that logs are not buffered in memory.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/77bb8a39-8992-402e-97be-4c5975295e1e)
code format

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: FALSE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to serialize the row batches in RPCs between BEs. TRUE indicates serializing the row batches, and FALSE indicates not serializing them.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/66da17d7-2f38-4934-b9f8-f11823786d6c)
code format

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+- **Description**: The directory and medium of the storage volume.
+  - Multiple volumes are separated by semicolons (`;`).
+  - If the storage medium is SSD, add `medium:ssd` at the end of the directory.
+  - If the storage medium is HDD, add ,medium:hdd at the end of the directory.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/3568d558-3d4f-497d-8657-78d689a165c3)
Wrap ",medium:hdd" in code format too. Also, I think the comma (,) in ",medium:hdd" is a typo and should be removed?

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: FALSE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to disable PageCache.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/be45e71d-5b18-4e54-8898-d55f653ab173)
Delete "A boolean value to control", so the sentence starts with "Whether".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+  - When PageCache is enabled, StarRocks caches the recently scanned data.
+  - PageCache can significantly improve the query performance when similar queries are repeated frequently.
+  - TRUE indicates disabling PageCache.
+  - The default value of this item has been changed from TRUE to FALSE since StarRocks v2.4.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/86322c9b-0dbc-448b-9256-e73250b642bf)
Wrap these in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: TRUE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to enable the token check. TRUE indicates enabling the token check, and FALSE indicates disabling it.
```
![image](https://github.com/StarRocks/starrocks/assets/57167462/0081c9d2-4444-4031-b0a3-ec4fa4cae4cc)
1. Remove "A boolean value to control", and make the sentence start with "Whether".
2. Wrap the parameter values "TRUE" and "FALSE" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: TRUE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to enable the pre-fetch of the query. TRUE indicates enabling pre-fetch, and FALSE indicates disabling it.
```
Remove "A boolean value to control", and make the sentence start with "Whether".

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: FALSE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to enable the synchronization of the tablet metadata. TRUE indicates enabling synchronization, and FALSE indicates disabling it.
```
1. Remove "A boolean value to control", and make the sentence start with "Whether".
2. Wrap the parameter values "TRUE" and "FALSE" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: 10
+- **Unit**: N/A
+- **Description**: The thread pool size for Routine Load on each BE. Since v3.1.0, this parameter is deprecated. The thread pool size for Routine Load on each BE is now controlled by the FE dynamic parameter max_routine_load_task_num_per_be.
```
Wrap "max_routine_load_task_num_per_be" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: FALSE
+- **Unit**: N/A
+- **Description**: A boolean value to control whether to enable the new storage format of the BITMAP type, which can improve the performance of bitmap_union. TRUE indicates enabling the new storage format, and FALSE indicates disabling it.
```
1. Remove "A boolean value to control", and make the sentence start with "Whether".
2. Wrap the parameter values "TRUE" and "FALSE" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: false
+- **Unit**: N/A
+- **Description**: Whether to enable Data Cache. TRUE indicates Data Cache is enabled, and FALSE indicates Data Cache is disabled.
```
Wrap the parameter values "TRUE" and "FALSE" in code format.

amber-create commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: 0
+- **Unit**: Bytes
+- **Description**: The maximum amount of data that can be cached on a single disk. For example, if you configure two disk paths for the block_cache_disk_path parameter and set the value of the block_cache_disk_size parameter as 21474836480 (20 GB), a maximum of 40 GB data can be cached on these two disks. The default value is 0, which indicates that only memory is used to cache data. Unit: bytes.
```
See to all parameter values starting from here till the end of this doc and make sure they are wrapped in code format.

DanRoscigno commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Unit**: s
+- **Default**: 300
+- **Description**: The maximum duration by which the metadata on the follower and observer FEs can lag behind that on the leader FE. Unit: seconds. If this duration is exceeded, the non-leader FE stops providing services.
```
```suggestion
- **Description**: The maximum duration by which the metadata on the follower and observer FEs can lag behind that on the leader FE. Unit: seconds. If this duration is exceeded, the non-leader FEs stops providing services.
```

DanRoscigno commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- Unit: LONG
+- Default: 107374182400
+- Description: The size of the largest partition for automatic collection to collect data. Unit: Byte. If a partition exceeds this value, full collection is discarded and sampled collection is performed instead.
```
```suggestion
- Description: The size, in bytes, of the largest partition for the automatic collection of statistics. If a partition exceeds this value, then sampled collection is performed instead of full.
```

DanRoscigno commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2042 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+#### scanner_thread_pool_queue_size
+
+- **Default:** 102,400 (Number of Tasks)
```
```suggestion
- **Default:** 102,400
```

DanRoscigno committed:
Apply suggestions from code review

wanpengfei-git commented:
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

DanRoscigno commented:
Thanks Amber.  I fixed many of the comments and will defer the rest until we split this page up.  For now I would like to:

1. get it merged and verify that search works
2. all of us adopt the use of Markdown H (`##`) tags instead of tables so that people can read the text and search can index it

evelynzhaojie commented on docs/administration/Configuration.md:
```diff
@@ -42,462 +42,2044 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
+
+- **Default**: 128
+- **Unit**: N/A
+- **Description**: Specifies the size of the Hedged Read thread pool on your HDFS client. The thread pool size limits the number of threads to dedicate to the running of hedged reads in your HDFS client. This parameter is supported from v3.0 onwards. It is equivalent to the dfs.client.hedged.read.threadpool.size parameter in the hdfs-site.xml file of your HDFS cluster.
```
maybe for parameters that have clear version info, we can add a row called "Supported in xxx" or add the version info to the parameter name, for example, ![image](https://github.com/StarRocks/starrocks/assets/98087056/369dd25d-3381-44ee-83c4-b53a2252c1dd). but I think the current change is great, we can add the version info later

DanRoscigno replied:
yes, Amber also had great suggestions for making this more meaningful. I will copy this and Amber's suggestions into a doc.

cc: @amber-create @evelynzhaojie

wanpengfei-git commented:
@Mergifyio backport branch-3.2

wanpengfei-git commented:
@Mergifyio backport branch-3.1
